src/shared/src/impls.rs

```diff
@@ -131,3 +133,16 @@ impl From<&StoredUserProfile> for UserProfile {
         }
     }
 }
+
+impl OisyUser {
+    #[must_use]
+    pub fn from_profile(user: &StoredUserProfile, principal: Principal) -> OisyUser {
+        OisyUser {
+            principal,
+            pouh_verified: user
+                .credentials
+                .contains_key(&CredentialType::ProofOfUniqueness),
```

Comment:
Good question.

The credential type means two things:
* The credential_type value of the credential.
* The type of credential Oisy considers that to be.

The enum `CredentialType` is about the second one: a way to differentiate credentials.

Yet, I implemented the formatting of the enum:

```
impl fmt::Display for CredentialType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            CredentialType::ProofOfUniqueness => write!(f, "ProofOfUniqueness"),
        }
    }
}
```

So that I could compare the value of the enum with the first `credential_type` in a credential when adding a credential.

I'm not so sure what you mean by "setting the enum variant representatives". Do you mean to add a value in the enum: `ProofOfUniqueness { id: u32 },`?

Thinking again about it, I'm also not so happy with the current solution being scalable to multiple issuers. The question is, is the effort to make it scalable worth the time? How many credentials will Oisy support? less than ten? hundreds?

I believe it will be more towards less than ten, we should be able to use an enum and different entries per each SupportedCredential.